merlin/systems/dag/ops/implicit.py

```diff
@@ -0,0 +1,119 @@
+        prev_output_schema: Schema = None,
+    ) -> Schema:
+        """Return the output schema representing the columns this operator returns."""
+        return Schema([ColumnSchema("ids", dtype="int64"), ColumnSchema("scores", dtype="float64")])
+
+    def compute_input_schema(
+        self,
+        root_schema: Schema,
+        parents_schema: Schema,
+        deps_schema: Schema,
+        selector: ColumnSelector,
+    ) -> Schema:
+        """Return the input schema representing the input columns this operator expects to use."""
+        return Schema([ColumnSchema("user_id", dtype="int64"), ColumnSchema("n", dtype="int64")])
+
+    def export(self, path, input_schema, output_schema, params=None, node_id=None, version=1):
+        """Export the class and related files to the path specified."""
+        node_name = f"{node_id}_{self.export_name}" if node_id is not None else self.export_name
+        version_path = pathlib.Path(path) / node_name / str(version)
+        version_path.mkdir(parents=True, exist_ok=True)
+        model_path = version_path / "model.npz"
+        self.model.save(str(model_path))
+        params = params or {}
+        params["model_module_name"] = self.model.__module__
+        params["model_class_name"] = self.model.__class__.__name__
+        return super().export(
+            path,
+            input_schema,
+            output_schema,
+            params=params,
+            node_id=node_id,
+            version=version,
+        )
+
+    @classmethod
+    def from_config(
+        cls, config: dict, model_repository="./", model_name=None, model_version=1
+    ) -> "PredictImplicit":
+        """Instantiate the class from a dictionary representation.
+
+        Expected structure:
+        {
+            "input_dict": str  # JSON dict with input names and schemas
+            "params": str  # JSON dict with params saved at export
+        }
+
+        """
+        params = json.loads(config["params"])
+
+        # load implicit model
+        model_module_name = params["model_module_name"]
+        model_class_name = params["model_class_name"]
+        model_module = importlib.import_module(model_module_name)
+        model_cls = getattr(model_module, model_class_name)
+        model_file = pathlib.Path(model_repository) / model_name / str(model_version) / "model.npz"
+        model = model_cls.load(str(model_file))
+
+        return cls(model)
+
+    def transform(self, df: InferenceDataFrame) -> InferenceDataFrame:
+        """Transform the dataframe by applying this operator to the set of input columns.
+
+        Parameters
+        -----------
+        df: InferenceDataFrame
+            A pandas or cudf dataframe that this operator will work on
+
+        Returns
+        -------
+        InferenceDataFrame
+            Returns a transformed dataframe for this operator"""
+        user_id = df["user_id"][0]
+        num_to_recommend = df["n"][0]
+        user_items = None
```

Comment:
could move this to the inputs if it's a common pattern for users to pass in. My understanding is that this mostly serves as a mechanism to filter out items already seen.

Reply:
I think this is ok - user_items can be left out if the filter_already_liked_items is False, and recalculate_user is also False (otherwise we need to now what the items the user has liked).